src/main/java/mekanism/client/render/RenderTickHandler.java

```diff
@@ -304,6 +311,25 @@ public void onBlockHover(DrawSelectionEvent.HighlightBlock event) {
             MultiBufferSource renderer = event.getMultiBufferSource();
             Camera info = event.getCamera();
             PoseStack matrix = event.getPoseStack();
+
+            ItemStack stack = player.getMainHandItem();
+
+            // Draw outlines for area mining blocks
+            if (!outliningArea && !player.isShiftKeyDown() && stack.getItem() instanceof ItemMekaTool tool && tool.isModuleEnabled(stack, MekanismModules.BLASTING_UNIT)) {
+                outliningArea = true;
+                Vec3 renderView = info.getPosition();
+
+                Set<BlockPos> blocks = ModuleBlastingUnit.findPositions(world, pos, rayTraceResult.getDirection(), tool.getModule(stack, MekanismModules.BLASTING_UNIT).getCustomInstance().getBlastRadius());
+                for (BlockPos block : blocks) {
+                    if (!ForgeHooksClient.onDrawHighlight(event.getLevelRenderer(), info, rayTraceResult, event.getPartialTicks(), matrix, renderer)) {
+                        event.getLevelRenderer().renderHitOutline(matrix, renderer.getBuffer(RenderType.lines()), minecraft.player, renderView.x, renderView.y, renderView.z, block, world.getBlockState(block));
+                    }
+                }
+                outliningArea = false;
+                event.setCanceled(true);
+                return;
```

Comment:
I feel like this may be better off skipping the position that is currently being done and then just allow it to fall through to the rest of our implementation without cancelling it (just to slightly reduce the number of events being fired).

I also at that point wonder if maybe the target block is a bounding block then it should instead skip the block we redirect the shape rendering to? (Not sure on this part will try to think about it more)

Reply:
The logic in ItemMekaTool probably needs changing to match too now that I think about it. Currently it is not area mining if the target is bounding, and not adding any bounding. But vein mining is only concerned with whether the target is bounding, which works with current area mine logic.